### ISSUE TITLE
BACnet error:invalid data-type

### ISSUE THREAD

**Issue by Pandacli** [MangoAutomation/BACnet4J#81]
Hi，all . I use  writePropertyRequest to write a new presentValue  , but unfortunately it response a wrong code 

invalid data-type. could you give us any advise?  thanks 

there is my code 
` WritePropertyRequest writePropertyRequest = new WritePropertyRequest(
                        objectList.get(i),
                        PropertyIdentifier.presentValue,
                        null, new Real(setVal), new UnsignedInteger(BaConfig.priority));
                ObjectType objectType = writePropertyRequest.getObjectIdentifier().getObjectType();
                ServiceFuture send = localDevice.send(remoteDevice, writePropertyRequest);

                ReadPropertyAck ack = localDevice.send(remoteDevice, writePropertyRequest)
                        .get();`
【project details 】
setValue = 1;
priority is 9 ; 

Device detail like this 
![image](https://user-images.githubusercontent.com/112819610/230886045-44f68d44-6326-4dc1-9c84-3d4228552921.png)

**Comment by kishorevenki** [comment]
HI,
Based on your above information, I noticed couple of points:

1. Since the Present_Value is not of array datatype, this value can be omitted. Mentioning Null is not correct. 
2. Since the Object Type is Binary Value, the datatype of PV is not Real, but BinaryPV. Hence the value can be new BinaryPV(1). 

Regards,
Kishore

**Comment by Pandacli** [comment]
Hi Kishore ,thanks for your advise 
and we fix it to change  the BinaryPV ,not a Real.


> HI, Based on your above information, I noticed couple of points:
> 
> 1. Since the Present_Value is not of array datatype, this value can be omitted. Mentioning Null is not correct.
> 2. Since the Object Type is Binary Value, the datatype of PV is not Real, but BinaryPV. Hence the value can be new BinaryPV(1).
> 
> Regards, Kishore

**Comment by kishorevenki** [comment]
Hope you will remove Null field as well. 

WritePropertyRequest writePropertyRequest = new WritePropertyRequest(
objectList.get(i),
PropertyIdentifier.presentValue,
new BinaryPV(setVal), new UnsignedInteger(BaConfig.priority));

**Comment by Pandacli** [comment]
> Hope you will remove Null field as well.
> 
> WritePropertyRequest writePropertyRequest = new WritePropertyRequest( objectList.get(i), PropertyIdentifier.presentValue, new BinaryPV(setVal), new UnsignedInteger(BaConfig.priority));

Yes,it work . I should use BinaryPV.active to control  Bacnet device . thank you so much
